python/cugraph/centrality/edge_betweenness_centrality_wrapper.pyx

```diff
@@ -0,0 +1,135 @@
+from cugraph.structure.graph import DiGraph, Graph
+from cugraph.structure.graph_new cimport *
+from cugraph.utilities.unrenumber import unrenumber
+from libc.stdint cimport uintptr_t
+from libcpp cimport bool
+import cudf
+import numpy as np
+import numpy.ctypeslib as ctypeslib
+
+def edge_betweenness_centrality(input_graph, normalized, weight, k,
+                                vertices, result_dtype):
+    """
+    Call betweenness centrality
+    """
+    # NOTE: This is based on the fact that the call to the wrapper already
+    #       checked for the validity of the implementation parameter
+    cdef GraphCSRView[int, int, float] graph_float
+    cdef GraphCSRView[int, int, double] graph_double
+
+    if not input_graph.adjlist:
+        input_graph.view_adj_list()
+
+    [offsets, indices] = graph_new_wrapper.datatype_cast([input_graph.adjlist.offsets, input_graph.adjlist.indices], [np.int32])
+
+    number_of_vertices= input_graph.number_of_vertices()
+    number_of_edges = len(indices)
+
+    df = cudf.DataFrame()
+    df['src'] = cudf.Series(np.zeros(number_of_edges, dtype=np.int32))
+    df['dst'] = indices.copy()
+    df['betweenness_centrality'] = cudf.Series(np.zeros(number_of_edges,
+                                               dtype=result_dtype))
+
+    cdef uintptr_t c_src_identifier = df['src'].__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_dst_identifier = df['dst'].__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_betweenness = df['betweenness_centrality'].__cuda_array_interface__['data'][0]
+
+    cdef uintptr_t c_offsets = offsets.__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_indices = indices.__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_weight = <uintptr_t> NULL
+    cdef uintptr_t c_vertices = <uintptr_t> NULL
+
+    if weight is not None:
+        c_weight = weight.__cuda_array_interface__['data'][0]
+
+    # FIXME: We could sample directly from a cudf array in the futur: i.e
+    #       c_vertices = vertices.__cuda_array_interface__['data'][0]
+    if vertices is not None:
+        c_vertices = np.array(vertices, dtype=np.int32).__array_interface__['data'][0]
+
+    c_k = 0
+    if k is not None:
+        c_k = k
+
+    # NOTE: The current implementation only has <int, int, float, float> and
+    #       <int, int, double, double> as explicit template declaration
+    #       The current BFS requires the GraphCSR to be declared
+    #       as <int, int, float> or <int, int double> even if weights is null
+    if result_dtype == np.float32:
+        graph_float = GraphCSRView[int, int, float](<int*> c_offsets,
+                                                    <int*> c_indices,
+                                                    <float*> NULL,
+                                                    number_of_vertices,
+                                                    number_of_edges)
+        # fixme: there might be a way to avoid manually setting the graph property
```

Comment:
That sounds similar to what we do in other wrappers. Do you / @Iroy30 have something in mind for this PR?

Reply:
Updated in 257b2f9 to use `get_graph_view` from https://github.com/rapidsai/cugraph/blob/branch-0.15/python/cugraph/structure/graph_new.pyx, one possibility would be to directly set the property there? @afender @Iroy30